src/main/java/uk/gov/hmcts/cmc/claimstore/services/InterestRateService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.InterestAmount;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+@Component
+public class InterestRateService {
+
+    public InterestAmount calculateRate(LocalDate fromDate, LocalDate toDate, double rate, double amount) {
```

Comment:
I would expect this service to return BigDecimal and then controller wraps value with `InterestAmount` which is RESTful data object. The `InterestRateService` should be used in all other places where we need to recalculate interest.